### PR TITLE
MWEB-2568 - Update check for empty string

### PR DESCRIPTION
**Pull request by bernardodsanderson**
Date will be coming as empty string instead of nil/null for Virtual Open House and Virtual Tour of Home, so we need to check against that.

Per Michael Wilson:
> The "" instead of null change has been shipped to production/alpha/staging for all MLSs and all listings with future open houses have been reloaded (or will be in the next couple minutes)

### CODE REVIEW

File: lib/spark_api/models/subresource.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def find_by_id(id, parent_id, arguments={})
 
       def parse_date_start_and_end_times(attributes)
         # Transform the date strings
-        unless attributes['Date'].nil?
+        unless attributes['Date'].nil? && attributes['Date'].empty?
           date = Date.strptime attributes['Date'], '%m/%d/%Y'
           ['StartTime','EndTime'].each do |time|
             next if attributes[time].nil?
```
Comment (wlmcewen) [inline lib/spark_api/models/subresource.rb]: I believe you'll want a similar empty check for `StartTime` and `EndTime` here. Based on @joshcom's API review earlier today, those fields have the same implication.

Comment (bernardodsanderson) [reply lib/spark_api/models/subresource.rb]: Will do, thanks Wade!

File: lib/spark_api/models/subresource.rb [hunk 2]
```diff
@@ -16,10 +16,10 @@ def find_by_id(id, parent_id, arguments={})
 
       def parse_date_start_and_end_times(attributes)
         # Transform the date strings
-        unless attributes['Date'].nil?
+        unless attributes['Date'].nil? && attributes['Date'].empty?
           date = Date.strptime attributes['Date'], '%m/%d/%Y'
           ['StartTime','EndTime'].each do |time|
-            next if attributes[time].nil?
+            next if attributes[time].nil? && attributes['Date'].empty?
```
Comment (wlmcewen) [inline lib/spark_api/models/subresource.rb]: Oops, I just noticed that both of these should be an `||`, unless I'm lost this late at night. The attribute can't be both null and empty :laughing:

File: lib/spark_api/models/subresource.rb [hunk 3]
```diff
@@ -16,10 +16,10 @@ def find_by_id(id, parent_id, arguments={})
 
       def parse_date_start_and_end_times(attributes)
         # Transform the date strings
-        unless attributes['Date'].nil?
+        unless attributes['Date'].nil? && attributes['Date'].empty?
```
Comment (wlmcewen) [inline lib/spark_api/models/subresource.rb]:
```
unless attributes['Date'].nil? || attributes['Date'].empty?
```

File: lib/spark_api/models/subresource.rb (hunk 2 above)
Comment (wlmcewen) [reply lib/spark_api/models/subresource.rb]:
```
next if attributes[time].nil? && attributes[time].empty?
```

**Comment by bernardodsanderson** [comment]
Thanks for the review @wlmcewen! My foggy late-night mind must have gotten the best of me on some of those.

**Pull request merged**